[BUG] Impossible to link more than one local module dependency one at a time.

<!--
Note: Please search to see if an issue already exists for your problem: https://github.com/npm/cli/issues
-->

### Current Behavior:
Running `:/moduleA $ npm link moduleB` seems to attempt to install moduleA before linking moduleB. If moduleA depends on another local moduleC, linking moduleB into moduleA is impossible.

### Expected Behavior:
Running `:/moduleA $ npm link moduleB` links moduleB into `moduleA/node_modules/moduleB`. (just like `npm@6` does)

### Steps To Reproduce:
Overview:
1. Create moduleB and moduleC that are not in npm registry.
2. Use moduleB and moduleC as a dependency in moduleA.
3. Execute `:/moduleA $ npm link moduleB`. It will result in failing to find moduleC in npm registry.

Details:
```
$ ll
total 0
drwxr-xr-x  3 tristan  wheel   96 Jan 27 08:24 moduleA
drwxr-xr-x  3 tristan  wheel   96 Jan 27 08:23 moduleB
drwxr-xr-x  3 tristan  wheel   96 Jan 27 08:23 moduleC
```
```
$ cat moduleA/package.json 
{
  "name": "modulea",
  "version": "1.0.0",
  "dependencies": {
    "moduleb": "1.0.0",
    "modulec": "1.0.0"
  }
}
```
```
$ cat moduleB/package.json 
{
  "name": "moduleb",
  "version": "1.0.0"
}
```
```
$ cat moduleC/package.json 
{
  "name": "modulec",
  "version": "1.0.0"
}
```
```
$ cd moduleB/
:/moduleB $ npm link

up to date, audited 3 packages in 476ms

found 0 vulnerabilities
```
```
$ cd ../moduleC
:/moduleC $ npm link

up to date, audited 3 packages in 778ms

found 0 vulnerabilities
```
Please note that in the below step we attempt to link moduleB, but it complains about not finding _**moduleC**_.
```
$ cd ../moduleA
:/moduleA $ npm link moduleb
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/modulec - Not found
npm ERR! 404 
npm ERR! 404  'modulec@1.0.0' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
```

For contrast, it works just fine using `npm@6`:
```
:/moduleA $ npm i -g npm@6

removed 57 packages, changed 194 packages, and audited 436 packages in 2s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
:/moduleA $ npm link moduleb
/moduleA/node_modules/moduleb -> /Users/tristan/.nvm/versions/node/v15.7.0/lib/node_modules/moduleb -> /moduleB
:/moduleA $ npm link modulec
/moduleA/node_modules/modulec -> /Users/tristan/.nvm/versions/node/v15.7.0/lib/node_modules/modulec -> /moduleC
```

### Environment:
MacOS
npm@7
<!--
ex.
- OS: Ubuntu 20.04
- Node: 13.14.0
- npm: 6.4.12
-->

Confirmed that `npm link moduleb modulec` is a valid workaround:

```
:/moduleA $ npm link moduleb
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/modulec - Not found
npm ERR! 404 
npm ERR! 404  'modulec@1.0.0' is not in the npm registry.
...
:/moduleA $ npm link modulec
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/moduleb - Not found
npm ERR! 404 
npm ERR! 404  'moduleb@1.0.0' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
...
:/moduleA $ npm link moduleb modulec

added 2 packages, and audited 5 packages in 635ms

found 0 vulnerabilities
```

Thank you.

--- a/lib/link.js
+++ b/lib/link.js
@@ -134,12 +134,14 @@ class Link extends ArboristWorkspaceCmd {
     // reify all the pending names as symlinks there
     const localArb = new Arborist({
       ...this.npm.flatOptions,
+      prune: false,
       log: this.npm.log,
       path: this.npm.prefix,
       save,
     })
     await localArb.reify({
       ...this.npm.flatOptions,
+      prune: false,
       path: this.npm.prefix,
       log: this.npm.log,
       add: names.map(l => `file:${resolve(globalTop, 'node_modules', l)}`),

--- a/test/lib/link.js
+++ b/test/lib/link.js
@@ -1,4 +1,5 @@
 const { resolve } = require('path')
+const fs = require('fs')
 
 const Arborist = require('@npmcli/arborist')
 const t = require('tap')
@@ -483,6 +484,55 @@ t.test('link pkg already in global space when prefix is a symlink', (t) => {
   link.exec(['@myscope/linked'], (err) => {
     t.error(err, 'should not error out')
   })
+})
+
+t.test('should not prune dependencies when linking packages', async t => {
+  const testdir = t.testdir({
+    'global-prefix': {
+      lib: {
+        node_modules: {
+          linked: t.fixture('symlink', '../../../linked'),
+        },
+      },
+    },
+    linked: {
+      'package.json': JSON.stringify({
+        name: 'linked',
+        version: '1.0.0',
+      }),
+    },
+    'my-project': {
+      node_modules: {
+        foo: {
+          'package.json': JSON.stringify({ name: 'foo', version: '1.0.0' }),
+        },
+      },
+      'package.json': JSON.stringify({
+        name: 'my-project',
+        version: '1.0.0',
+      }),
+    },
+  })
+  npm.globalDir = resolve(testdir, 'global-prefix', 'lib', 'node_modules')
+  npm.prefix = resolve(testdir, 'my-project')
+  reifyOutput = () => {}
+
+  const _cwd = process.cwd()
+  process.chdir(npm.prefix)
+
+  await new Promise((res, rej) => {
+    link.exec(['linked'], (err) => {
+      if (err)
+        rej(err)
+      res()
+    })
+  })
+
+  t.ok(
+    fs.statSync(resolve(testdir, 'my-project/node_modules/foo')),
+    'should not prune any extraneous dep when running npm link'
+  )
+  process.chdir(_cwd)
 })
 
 t.test('completion', async t => {

